ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                poFeature->GetFieldAsInteger64(iField);
+            if (MM_SecureCopyStringFieldValue(
+                    &hMMFeature.pRecords[0].pField[iField].pDinValue,
+                    poFeature->GetFieldAsString(iField),
+                    &hMMFeature.pRecords[0].pField[iField].nNumDinValue))
+                return OGRERR_NOT_ENOUGH_MEMORY;
+            hMMFeature.pRecords[0].pField[iField].bIsValid = 1;
+        }
+        else if (eFType == OFTReal)
+        {
+            hMMFeature.nNumMRecords = max_function(hMMFeature.nNumMRecords, 1);
+            hMMFeature.pRecords[0].nNumField = nNumFields;
+            if (MMResizeMiraMonFieldValue(&(hMMFeature.pRecords[0].pField),
+                                          &hMMFeature.pRecords[0].nMaxField,
+                                          hMMFeature.pRecords[0].nNumField,
+                                          MM_INC_NUMBER_OF_FIELDS,
+                                          hMMFeature.pRecords[0].nNumField))
+                return OGRERR_NOT_ENOUGH_MEMORY;
+
+            hMMFeature.pRecords[0].pField[iField].dValue =
+                poFeature->GetFieldAsDouble(iField);
+            if (MM_SecureCopyStringFieldValue(
+                    &hMMFeature.pRecords[0].pField[iField].pDinValue,
+                    poFeature->GetFieldAsString(iField),
+                    &hMMFeature.pRecords[0].pField[iField].nNumDinValue))
+                return OGRERR_NOT_ENOUGH_MEMORY;
+            hMMFeature.pRecords[0].pField[iField].bIsValid = 1;
+        }
+        else
+        {
+            CPLError(CE_Failure, CPLE_NotSupported,
+                     "MiraMon: Field type %d not processed by MiraMon\n",
+                     eFType);
+            return OGRERR_UNSUPPORTED_GEOMETRY_TYPE;
+        }
+    }
+
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                             GetLayerDefn()                               */
+/*                                                                          */
+/****************************************************************************/
+OGRFeatureDefn *OGRMiraMonLayer::GetLayerDefn()
+{
+    return poFeatureDefn;
+}
+
+/****************************************************************************/
+/*                             GetExtent()                                  */
+/*                                                                          */
+/*      Fetch extent of the data currently stored in the dataset.           */
+/*      The bForce flag has no effect on SHO files since that value         */
+/*      is always in the header.                                            */
+/****************************************************************************/
+
+OGRErr OGRMiraMonLayer::GetExtent(OGREnvelope *psExtent, int bForce)
+
+{
+    if (phMiraMonLayer)
+    {
+        psExtent->MinX = phMiraMonLayer->TopHeader.hBB.dfMinX;
```

Comment:
I've noticed  "ogrinfo autotest/ogr/data/miramon/Arcs/EmptyArcs/Empty_ARC.arc -al" return a garbage extent. I suspect that should test here that the number of features is not 0. If it is zero, then return OGRERR_FAILURE

Reply:
Indeed: After the change... no garbage
Feature Count: 0
Layer SRS WKT: